### DISCUSSION TITLE
Making links to other RSE groups

### DISCUSSION THREAD

**Discussion by metazool** [NERC-CEH/rse_group#15] [General]
Robin suggested networking a bit more with other RSE groups that are aligned by organisation type and function (BAS, BGS etc)

There are also academic groups whose work is close to UKCEH in theme - at the People in Data meeting i saw (but didn't speak with) Kate Court at University of Newcastle, their workload is environment focused?

https://rse.ncldata.dev/team/kate-court

**Comment by metazool** [comment]
https://rse.ncldata.dev/reports/nclrse-2023-public.pdf - Newcastle RSE group's public report for 2023 including a project index - they're a big team!

**Comment by longr** [comment]
We could look at a "regional" RSE grouping through SocRSE.   For this I was thinking smaller non-University RSE groups.   
But more contacts with environmental focussed RSEs or sub-groups could be useful.

**Comment by KateCourt** [comment]
Thanks for linking up @metazool, for reference, we work across the whole Uni, not just environment stuff! Happy to chat more if it's helpful :)

**Comment by metazool** [comment]
I think BGS have more of an RSE "capability" than a formalised group. It weights towards the DevOps / infrastructure side of activity. The membership is pretty well represented by the contributor list to the [etlhelper](https://github.com/BritishGeologicalSurvey/etlhelper) library, and @volcan01010 is BGS's "RSE Capability Lead"

**Comment by mattjbr123** [comment]
I beleive there are a couple of RSEs (maybe more?) being hired/are already hired specifically for the FDRI project, and others in RSE-like roles scattered about the organisation. It would be nice to draw them in a bit too, though not sure how just yet. Maybe something for when we are more visible/known ourselves.

**Comment by metazool** [comment]
Discussion in standup about possible topic of "how we are approaching Agile" for a forum/meeting across groups

https://www.itservices.manchester.ac.uk/research/meet/ - massive group in Manchester, @longr picks out @aharwood2 as someone specifically with a lot of groundwork on more "standard" development practises for research software who it would be good to learn from.

I've had such mixed experiences with process-heavy Scrum; a regular cycle of demo / Q&A sessions, a place to quickly check what's in shared scope, and a rhythm of meetings kept informal, and otherwise run things on asynchronous goodwill like an open source project, that's honestly enough for me. This probably belongs in [another thread](#16 )! I think it would be a dry-but-beneficial topic for a wider forum

**Comment by aharwood2** [comment]
Thanks for the shoutout @metazool and @longr -- in Manchester we have recently rolled out an "agile methods handbook" with complementary training to convert all our project delivery to using a "lightweight" version of Scrum which hopefully is flexible enough that it can be applied whether academic engagement is high or low, projects are big or small, long or short. We've been doing it in anger for a few months now and results generally seem to be quite positive.

**Comment by metazool** [comment]
It's friendly of you to participate in our forum, thankyou @aharwood2 ! It would be a service if you were able to publically link to the "agile methods handbook" that the Manchester group have developed?

It's slightly tangential, but as a new RSE group we are also involved in co-design of a project acceptance process that defines how we spend our time. There's a set of high-level principles describing what kind of work we should accept, and a committee that has the final say; but quite a lot of detail to be worked out in between, about how this can be done in a [FAIR](https://www.go-fair.org/fair-principles/) and [CARE](https://www.gida-global.org/care) compatible way.

There's bound to be a lot of difference in cultural setting across different RSE groups - how project allocations work, what "excellence" looks like, whether a group is more embedded in IT services or development / data science. There's bound to be a lot of ground that's the same, too. I'll try to attend the RSE Leaders meeting before RSECon in Newcastle and maybe bring this as a topic. Perhaps there are records of the past times this discussion has happened?

**Comment by longr** [comment]
I know that @thomaszwagerman is an RSE in [BAS](https://www.bas.ac.uk/), I cannot find a list or more details on their team. @colinsauze is also an senior RSE at [NOC](https://www.noc.ac.uk).  Do either NOC or BAS have formal teams or is it more RSE capability?  Either way we would like to chat with you and your colleagues.

**Reply by thomaszwagerman** [reply]
Thank you for tagging me in this discussion Robin.

At BAS I'm part of the Digital Innovation Team, we're currently a team of 5 RSEs but may include other research infrastructure roles in the future, hence I suppose not strictly an "RSE group". We are also a new team, so I think face similar challenges to some of what I'm reading above. My team is led by @JimCircadian, so I will tag him here and he can share more about that ;).

I'm always keen for closer collaboration, as we're finding out on BOOST-EDS Robin, there are a lot of overlapping areas of interest where it would be useful to share experiences, work together and exchange ideas (pipelines, DTs, cloud formats, DRI in general). Also, as "RSEs in NERC" it would be interesting to explore whether there are any topics which we might collectively want to take a lead on when it comes future funding of NERC EDS projects.

Jo, I saw you attended People in Data as well - I meant to say hello but wasn't sure how to do that "hybridly". I used to be based at CEH Edinburgh as an Environmental Data Scientist working with David Leaver. We discussed RSE roles many times (in fact he introduced me to the concept in the first place, thanks David!), so I'm really happy to see there is now a formal RSE group.

**Reply by thomaszwagerman** [reply]
PS I am on annual leave this week, so apologies if you are waiting for a response from me on other communication channels :grin: - couldn't help myself taking part in this discussion though!

**Reply by JimCircadian** [reply]
@longr @metazool Happy for you to get in contact with me direct or via the team (@thomaszwagerman is a great start, I also have touchpoints into FDRI and the general Digital Twins / DRI efforts across NERC centres.) My email is on the BAS website. Also worth looking in [here](https://society-rse.org/community/rse-groups/) to locate RSE groups that might be of interest. Nice one for reaching out!

**Comment by colinsauze** [comment]
I'm not sure what you define as a formal team vs RSE Capability. At NOC we currently have 3 RSEs (soon to be 7) who are part of a centralised team that can be hired out to projects which need them and also develop some centralised platforms and run training. 

You might be interested in attending the "Finding the weather and climate RSE community" BoF session at RSECon, https://virtual.oxfordabstracts.com/#/event/49081/session/118382 where there will hopefully be RSEs from a wide range of organisations who work on weather and climate related research. One thing that's up for discussion at this session will be whether we should form a special interest group within the RSE society that can facilitate networking.

**Reply by metazool** [reply]
Thank you, that BoF looks promising, the majority of our group will be in-person at RSECon. If you started a special interest group maybe broaden it to include environmental science in general? 

The "RSE Capability" term comes from BGS where @volcan01010 is serving as "RSE Capability Lead" but there isn't a centralised team with defined edges or a distinct way of "hiring" them, more of a grouping of developers in the central Informatics service that do RSE-like collaboration work (as well as dedicated coders within compute-heavy science groups like Geomagnetism)

**Comment by longr** [comment]
Thanks for responding @colinsauze. Glad to see the team will be expanding.   By RSE capacity I mean that there are RSEs in the organisation, but they are not part of a central team.    I won't be able to make RSECon this year, but most of the CEH team will be there.   Not sure if we would quite fit in a weather and climate SIG, but I especially would be interested in a NERC / Environmental SIG, or even a Small Institutes SIG as I think there is a lot to learn in both.

**Comment by metazool** [comment]
@longr  if you (or others who were in those BOOST sessions) could share brief notes on topics and links shared with @thomaszwagerman and others from BAS, in this Discussion, that would be much appreciated by those of us on other sites!

**Comment by volcan01010** [comment]
Hi all,

Thanks for tagging me, @metazool.

A BGS we have a large Informatics division which makes up nearly 20% of the organisation.  Part of BGS' purpose is to be a repository of geological data so it has a long history of database and data management staff.  The Informatics group contains database admins and developers, but also GIS specialists and software developers working on front- and back-end tools for sharing and managing the data.  These developers are a shared resource and their time can be bought be science teams e.g. to write a website for sharing their data or building software to do analysis.  Much of this work falls into the RSE category, even if it isn't explicitly called that within BGS.

I came to BGS as a scientist-turned-software-developer who wrote a lot of Python for my research.  @metazool, myself and others have worked to expand the use of Python within BGS, both within and outside Informatics.  It was clear that there is a huge demand among scientists to write Python in their own work, so we worked on training courses and drop-in clinics.  I have also been working with our central IT division to look at things like securely using open source software.

In a recent restructure, a set of "capabilities" were created within Informatics to make it clearer to the rest of BGS what skills we have that we could help with.  It was great to see RSE recognised as a discipline by the organisation.  I am the capability lead.  The main part of the role, as I see it, is at the helping scientists write their own code end of the scale.  There are many scientists in this position, so every little thing that helps them will make a big difference to the overall organisation.  Development work that falls under that RSE bracket includes projects where a developer works on code that was initially written by scientists to help them to improve it.

I'm going to the RSE conference in Newcastle next month, so hopefully I can meet up with some of you there.  I'm giving a talk called "SQLite for Scientists" on the Wednesday afternoon.

Re: the discussions about Agile above, note that there are some sessions about Agile for RSEs on the Wednesday morning.  At BGS, the SCRUM model doesn't work for us as Informatics staff all work on multiple projects and in different teams so we don't have the continuity that is required.  Instead we track issues on the GitLab boards in an approximately Kanban way, but I think that we get more benefit from the development practices side e.g. pairing, testing, automated deployments, getting rapid user feedback than the ceremonies.

Hopefully I'll see some of you next month,
John

**Reply by mattjbr123** [reply]
Thanks for this @volcan01010 it was great to chat to you at RSECon, and to reread this post! Always nice to put a face to some text!!

**Reply by volcan01010** [reply]
Great to meet you, too.  We should have a BGS / CEH meetup when you are next in the Edinburgh office.

**Comment by metazool** [comment]
via @iulusoy on handbooks https://zenodo.org/records/10867903 

cf #40

**Reply by iulusoy** [reply]
This is the white paper where we have summarized the mission of our RSE department, and the services we offer, together with examples. Project selection and time allocation is tricky, becoming more so with a growing group and demand. We are also working on onboarding material for new RSEs, and are trying out new templates for project documentation; using GitHub for everything.
